src/main/java/com/shapesecurity/salvation/data/Policy.java

```diff
@@ -341,10 +401,32 @@ public void intersectDirective(@Nonnull Directive<? extends DirectiveValue> dire
 		if (!(directive instanceof DefaultSrcDirective)) {
 			this.expandDefaultSrc();
 		}
-		this.intersectDirectivePrivate(directive);
+		Set<Directive> expandedDirectives = expandDirective(directive);
+		expandedDirectives.forEach(x -> this.intersectDirectivePrivate(x));
 		this.optimise();
 	}
 
+	private static Set<Directive> expandDirective(@Nonnull Directive<? extends DirectiveValue> directive) {
```

Comment:
Are you sure this is correct? If a CSP has `script-src 'https://a.com'` and also `script-src-elem 'https://b.com'`, would `<script src="https://b.com"></script>` be allowed? This (if I'm reading it correctly) implies that it would not, but I'm not sure what the spec says.